BMI323 sfusion implementation

jojos38 opened the pull request:
Working BMI323 at 200hz gyro and 200hz accel

Wasn't tested with two sensors as I don't have any but works fine on one sensor

TheDevMinerTV commented on src/defines.h:
```diff
@@ -26,9 +26,9 @@
-#define IMU IMU_BNO085
-#define SECOND_IMU IMU
-#define BOARD BOARD_SLIMEVR
+#define IMU IMU_BMI323
+#define SECOND_IMU IMU_BMI323
+#define BOARD BOARD_CUSTOM
```
Please don't change these, the firmware should always compile for the official trackers.

TheDevMinerTV commented on src/defines.h:
```diff
@@ -149,6 +149,24 @@ IMU_DESC_ENTRY(IMU_BMP160, PRIMARY_IMU_ADDRESS_ONE, IMU_ROTATION, PIN_IMU_SCL, P
 //  #define LED_INVERTED false
 #elif BOARD == BOARD_CUSTOM
   // Define pins by the examples above
+  #define PIN_IMU_SDA 4
+  #define PIN_IMU_SCL 5
+  #define PIN_IMU_INT 14
+  #define PIN_IMU_INT_2 12
+  #define PIN_BATTERY_LEVEL A0
+  #define LED_PIN 15
+  #define LED_INVERTED false
+  // The Wemos D1 mini has an internal voltage divider that I don't have on my custom board
+  // Therefor, BATTERY_SHIELD_RESISTANCE is set to 0 and SHIELD_R1 and R2 are adjusted instead
+  #ifndef BATTERY_SHIELD_RESISTANCE
+    #define BATTERY_SHIELD_RESISTANCE 0
+  #endif
+  #ifndef BATTERY_SHIELD_R1 
+    #define BATTERY_SHIELD_R1 36
+  #endif
+  #ifndef BATTERY_SHIELD_R2
+    #define BATTERY_SHIELD_R2 123.5
+  #endif
```
these should also be removed.
```suggestion
```

jojos38 commented:
Done, I forgot to revert these sorry

Eirenliel reviewed (approved):
The libraries say that it's your copyright, but they look like they were taken from the Bosch's sources. What's the library source?

jojos38 commented:
> The libraries say that it's your copyright, but they look like they were taken from the Bosch's sources. What's the library source?

I rewrote an optimized the code of the library, the library was originally written without OOP, I rewritten (only the required parts) with OOP and also optimized some parts to reduce memory usage and increase speed slighty
